src/pages/calendar.tsx

```diff
@@ -1,32 +1,398 @@
+  newAbsence?: Omit<Event, 'id'>;
+}
+
+interface FetchEventsResponse {
+  absences: Event[];
+}
+
+function CalendarView() {
+  const [events, setEvents] = useState<Event[]>([]);
+  const [value, setValue] = useState<Value>(new Date());
+  const [isFormOpen, setIsFormOpen] = useState(false);
+  const [formDate, setFormDate] = useState<Date | null>(null);
+  const [view, setView] = useState('month');
+
+  useEffect(() => {
+    FetchEvents();
+  }, []);
+
+  const FetchEvents = async () => {
+    const res = await fetch('/api/absence');
+    const data: FetchEventsResponse = await res.json();
+    setEvents(
+      data.absences.map((event) => ({
+        ...event,
+        lessonDate: new Date(event.lessonDate),
+      }))
+    );
+  };
+
+  const TileContent = ({ date, view }: { date: Date; view: string }) => {
+    if (view === 'month') {
+      const dayEvents = events.filter(
+        (event) => event.lessonDate.toDateString() === date.toDateString()
+      );
+      return (
+        <Box position="relative" height="100%" width="100%">
+          <Button
+            size="xs"
+            colorScheme="green"
+            onClick={() => onAddButtonClick(date)}
+            position="absolute"
+            top="2px"
+            right="2px"
+            zIndex="1"
+            minW="20px"
+            height="20px"
+            p="0"
+          >
+            +
+          </Button>
+          <VStack
+            spacing={1}
+            pt="24px"
+            pb="2px"
+            px="2px"
+            align="stretch"
+            height="100%"
+            width="100%"
+            overflowY="auto"
+          >
+            {dayEvents.map((event, index) => (
+              <Box
+                key={event.id || index}
+                bg="blue.100"
+                p={1}
+                borderRadius="sm"
+                boxShadow="sm"
+                _hover={{ bg: 'blue.200', cursor: 'pointer' }}
+              >
+                <Text fontSize="xs" fontWeight="bold" isTruncated>
+                  {event.subject}
+                </Text>
+                <Button
+                  size="xs"
+                  colorScheme="red"
+                  mt={1}
+                  onClick={(e) => {
+                    e.stopPropagation();
+                    HandleEventDelete(event.id);
+                  }}
+                >
+                  Delete
+                </Button>
+              </Box>
+            ))}
+          </VStack>
+        </Box>
+      );
+    }
+    return null;
+  };
+
+  const onAddButtonClick = (date: Date) => {
+    setFormDate(date);
+    setIsFormOpen(true);
+  };
+
+  const HandleEventDelete = async (id: number) => {
+    try {
+      const response = await fetch('/api/absence', {
+        method: 'DELETE',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({ id }),
+      });
+      if (response.ok) {
+        setEvents((prevEvents) =>
```

Comment:
I think using the GET endpoint is better than filtering out the absences with the deleted id